aerich/migrate.py

```diff
@@ -189,21 +190,19 @@ def _add_operator(cls, operator: str, upgrade=True, fk_m2m_index=False) -> None:
 
     @classmethod
     def _handle_indexes(cls, model: Type[Model], indexes: List[Union[Tuple[str], Index]]) -> list:
-        ret: list = []
+        if index_classes := set(index.__class__ for index in indexes if isinstance(index, Index)):
+            for index_cls in index_classes:
+                if index_cls(fields=("id",)) != index_cls(fields=("id",)):
 
-        def index_hash(self) -> str:
-            h = hashlib.new("MD5", usedforsecurity=False)  # type:ignore[call-arg]
-            h.update(
-                self.index_name(cls.ddl.schema_generator, model).encode()
-                + self.__class__.__name__.encode()
-            )
-            return h.hexdigest()
+                    def _hash(self) -> int:
+                        return hash((tuple(sorted(self.fields)), self.name, self.expressions))
 
-        for index in indexes:
-            if isinstance(index, Index):
-                index.__hash__ = index_hash  # type:ignore[method-assign,assignment]
-            ret.append(index)
-        return ret
+                    def _eq(self, other) -> bool:
+                        return type(self) is type(other) and hash(self) == hash(other)
```

Comment:
This is an incorrect implementation of `__eq__`.  Hashes happen to have collisions.